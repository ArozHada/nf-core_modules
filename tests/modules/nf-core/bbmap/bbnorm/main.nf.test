nextflow_process {

    name "Test Process BBMAP_BBNORM"
    script "modules/nf-core/bbmap/bbnorm/main.nf"
    process "BBMAP_BBNORM"
    config "tests/modules/nf-core/bbmap/bbnorm/nextflow.config"
    tag "bbmap_bbnorm"

    test("Sarscov2 Illumina SE") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test', single_end:true ], // meta map
                            file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true)
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path("$outputDir/bbmap/versions.yml"),
                                  path("$outputDir/bbmap/test.fastq.gz")).match("single_end") },
                { assert new File("$outputDir/bbmap/test.bbnorm.log").exists() }
            )
        }
    }

    test("Sarscov2 Illumina PE") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [ 
                            [ id:'test', single_end:false ], // meta map
                            [  
                             file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                             file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true) 
                            ]
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path("$outputDir/bbmap/versions.yml"),
                                  path("$outputDir/bbmap/test_1.nm.fastq.gz"),
                                  path("$outputDir/bbmap/test_2.nm.fastq.gz")).match("paired_end") },
                { assert new File("$outputDir/bbmap/test.bbnorm.log").exists() }
            )
        }
    }
}
