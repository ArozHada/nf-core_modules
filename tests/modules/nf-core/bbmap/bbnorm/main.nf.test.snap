{
    "single_end": {
        "content": [
            "versions.yml:md5,90b331a3702d052d61e0a14189e3fc01",
            "test.fastq.gz:md5,709872fc2910431b1e8b7074bfe38c67"
        ],
        "timestamp": "2023-04-08T02:37:52+0000"
    },
    "paired_end": {
        "content": [
            "versions.yml:md5,90b331a3702d052d61e0a14189e3fc01",
            "test_1.nm.fastq.gz:md5,a3a03cfae9f7afcba34a7b5a5875d4ea",
            "test_2.nm.fastq.gz:md5,8775e4088fa9ed85cf9e837cb27ec207"
        ],
        "timestamp": "2023-04-08T02:37:52+0000"
    }
}